### PR TITLE
chore(auth): reorganize auth javascript

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR does not implement any functional changes. It distributes the `auth.js` file into a new `reader-activation-auth` directory for better maintainability.

### How to test the changes in this Pull Request:

No changes should be visible, go through the authentication flow and confirm it works without issues.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by laurelfulford — COMMENTED** [review]
Hi @miguelpeixe! I ran through a few forms of authentication and mostly got through it error free -- I did run into one issue to start, but I could also recreate it with the current plugin release, and after I tried a few times it stopped happening. 

These are the tests that went fine:
* I tried creating a new account, and authenticating through the email link: no issues.
* I tried logging into that account with a code: no issues.
* I tried creating a password, and logging in with it: no issues.
* I tried resetting the password, and logging in with the new one: no issues.

Then I tried logging in with Google. This is my first time setting it up on a test site so I may have screwed something up. It seemed to mostly work, except after I confirmed sharing data, the modal showed these weird errors -- first is on my localhost, second on a JN site: 

Screen Shot 2024-01-23 at 2 18 44 PM
Screen Shot 2024-01-23 at 2 18 52 PM
Screen Shot 2024-01-23 at 2 23 07 PM

I got the exact same thing when I updated the Newspack Plugin to the current release on the Jurassic Ninja site. Despite the errors, when I closed the modal/refreshed the window, the account was created just fine on both sites. 

After testing a few times, it started working fine for me, so I think it may have been a temporary fluke?

I edited this comment after getting it successfully working, but left the details in here in case it sounds like something you've run into before! I'm going to mark this as approved though, since it works, and when I experienced the issue, I was running into the same thing with the current release, not just this PR.

**Review by laurelfulford — APPROVED** [review]

**Comment by miguelpeixe** [comment]
Thank you for thoroughly testing this, Laurel! I was also able to reproduce the issue with the Google OAuth and filed it in 1205234045751551-as-1206430738999629.

I was also able to get it working on the second time. I wonder if it just crashes if the account is being created (and not just signing in to an existing account).

I'll merge this one because it's an existing issue that this PR doesn't touch.

**Pull request merged**
